be/src/storage/lake/rowset_update_state.cpp

```diff
@@ -488,7 +487,7 @@ Status RowsetUpdateState::rewrite_segment(const TxnLogPB_OpWrite& op_write, cons
     std::vector<bool> need_rename(rowset_meta.segments_size(), true);
     for (int i = 0; i < rowset_meta.segments_size(); i++) {
         const auto& src_path = rowset_meta.segments(i);
-        const auto& dest_path = op_write.rewrite_segments(i);
+        const auto& dest_path = gen_segment_filename(txn_id);
```

Comment:
Generating a new segment name each time you publish does not guarantee tablet metadata retry idempotency, and if you don't guarantee idempotency, you don't guarantee cache consistency.